### DISCUSSION TITLE
Auto Update Mods after steam update

### DISCUSSION THREAD

**Discussion by tobasium** [GameServerManagers/LinuxGSM#3502] [Q&A] [answered]
Hi,

is it possible to run the ./rustserver mods-update command after a update? 
Is it possible to put the comand in the update script?

Thanks for help did not fijnd a solution.

**Comment by visdmin** [comment, marked as answer]
Hello,

There is at least two approaches that came into my mind. 

-----

1. Creating your own simple script that runs both commands one after other.
     - I would strongly suggest doing this.
2. Actually modifying LGSM functions to chain these functionalities together.
    - I would not recommend doing this because you could cause unwanted side effects, and updating LGSM could remove your modifications.

-----

**Creating your own simple script that executes both command one after other.**
Here is script that i made, you have to edit the `RUST_SERVER_PATH` variable so that script can find your lgsm `rustserver` executable.
How it works:
- Script runs `./rustserver update` and captures that status of the command (success or failure)
    - If server update fails the script informs the user with message, and stops.
    - If server update command is successful the script continues to updating mods
- Script runs  ./rustserver mods-update` and captures the status of the command (success of failure)
    - Script informs the user if mods update command was successful and stops.
    
If this solution does not fit your needs, please let me know, so we can find the correct solution for you.

How to use the script:
**NOTE**: all the actions are done with the same user that you use for running your rust server.

1. Upload the script to your lgsm installation directory and make it executable:
    - I use `nano` text editor to copy-and-paste the script to my server, please use what ever method that suits you:
    ```shell
    nano /home/lgsm-daemon/custom_rust_update.sh
    ```
    - I copy paste the script into the newly created file
    - I change the `RUST_SERVER_PATH` variable to point to my `rustserver`:
    ```log
    RUST_SERVER_PATH="/opt/my-custom-directory/rust-server/rustserver"
    ```
    - save and exit with `CTRL + X` -> `Y` -> `ENTER`

2.  Run the script manually
    - Now you can run the script located next to your `rustserver` manually to update server and mods with single command:
    ```shell
    ./custom_rust_update.sh
    ```
    - or 
    ```shell
    /opt/your-path/to/custom_rust_update.sh
    ```
If you would like to automate your server updates with cronjobs let me know, that is also possible.

-----

`custom_rust_update.sh`: https://gist.github.com/visdmin/bcad1b404bde04483de1f86bbd96bbf7
